test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+validate_and_cleanup_ext_block(WT_EXT *ext)
+{
+    REQUIRE(ext != nullptr);
+    REQUIRE(ext->depth != 0);
+    __wt_free(nullptr, ext);
+}
+
+void
+validate_and_cleanup_size_block(WT_SIZE *size)
+{
+    REQUIRE(size != nullptr);
+    __wt_free(nullptr, size);
+}
+
+void
+cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_EXT *curr = bms->ext_cache;
+    WT_EXT *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_EXT *curr = bms->ext_cache;
+
+    REQUIRE(bms->ext_cache_cnt == expected_items);
+    for (; i < expected_items; i++) {
+        REQUIRE(bms->ext_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+    cleanup_ext_list(bms);
+}
+
+void
+cleanup_size_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_SIZE *curr = bms->sz_cache;
+    WT_SIZE *tmp;
+    while (curr != NULL) {
+        tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_cleanup_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    int i = 0;
+
+    REQUIRE(bms != nullptr);
+    WT_SIZE *curr = bms->sz_cache;
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
+    for (; i < expected_items; i++) {
+        REQUIRE(bms->sz_cache != nullptr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+    cleanup_size_list(bms);
+}
+
+TEST_CASE("Block session: __block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_EXT *ext;
+
+    ext = nullptr;
+    __wt_random_init(&session->getWtSessionImpl()->rnd);
+
+    REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+    validate_and_cleanup_ext_block(ext);
+}
+
+TEST_CASE("Block session: __block_ext_prealloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero extent blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_cleanup_ext_list(bms, 0);
+    }
+
+    SECTION("Allocate one ext blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_cleanup_ext_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple ext blocks")
```

Comment:
Why not 3 and not 1K? Or even 100K?

Reply:
IMO, i chose 3 to make sure that we are testing the cache. I don't really see much benefit from increasing from 3 other than the fact that we are stress testing the function. Let me know if you disagree and see benefit from increasing it.